wiki/Client/File_formats/osr_(file_format)/de.md

```diff
@@ -0,0 +1,101 @@
+# .osr (Dateiformat)
+
+**.osr** ist ein Dateiformat mit Informationen über ein osu!-Replay. Für die Verwendung muss sich die zum Replay gehörende Beatmap im Ordner "Songs" befinden.
+
+## Datentypen
+
+| Name | Länge in Bytes | Beschreibung |
+| :-- | :-- | :-- |
+| Byte | 1 | Ein einzelner 8-Bit-Wert. |
+| Short | 2 | Ein 2-Byte-Wert im Little-Endian-Format. |
+| Integer | 4 | Ein 4-Byte-Wert im Little-Endian-Format. |
+| Long | 8 | Ein 8-Byte-Wert im Little-Endian-Format. |
+| ULEB128 | Variabel | Eine Ganzzahl mit variabler Länge. Siehe [ULEB128](https://en.wikipedia.org/wiki/LEB128). |
+| String | Variabel | Besteht aus drei Teilen. Ein einzelnes Byte, das entweder 0x0b (11 im Dezimalsystem) oder 0x00 ist, abhängig davon, ob die nächsten zwei Teile vorhanden sind oder nicht. Bei einem Wert von 0x0b bestehen die nächsten zwei Teile aus einem ULEB128, der die Länge in Byte des folgenden Strings definiert, und der String selbst, kodiert in UTF-8. Siehe [UTF-8](https://de.wikipedia.org/wiki/UTF-8). |
+
+## Format
+
+Byte-Offsets sind in dieser Tabelle nicht enthalten, da die Werte unterschiedlich lang sein können.
+
+| Datentyp | Beschreibung |
+| :-- | :-- |
+| Byte | Spielmodus des Replays (0 = osu!, 1 = osu!taiko, 2 = osu!catch, 3 = osu!mania) |
+| Integer | Version des Spiels, in der das Replay erstellt wurde (z. B. 20131216) |
+| String | MD5-Hash der osu!-Beatmap |
+| String | Spielername |
+| String | MD5-Hash des osu!-Replays (umfasst bestimmte Angaben des Replays) |
+| Short | Anzahl der 300er |
+| Short | Anzahl der 100er in osu!, 150er in osu!taiko, 100er in osu!catch, 100er in osu!mania |
+| Short | Anzahl der 50er in osu!, kleine Fruits in osu!catch, 50er in osu!mania |
+| Short | Anzahl der Gekis in osu!, Max 300er in osu!mania |
+| Short | Anzahl der Katus in osu!, 200er in osu!mania |
+| Short | Anzahl der Misses |
+| Integer | Gesamtpunktzahl (wird auf der Ergebnisanzeige dargestellt) |
+| Short | Höchste Combo (wird auf der Ergebnisanzeige dargestellt) |
+| Byte | Perfect/Full Combo (1 = keine Misses und keine Sliderbreaks sowie keine zu früh beendeten Slider) |
+| Integer | Verwendete Mods. Siehe unten für eine Liste der Modwerte. |
+| String | Lebensleiste: Komma-separierte Paare u/v, wobei u die Zeit in Millisekunden nach Beginn des Songs ist und v eine Gleitkommazahl zwischen 0 und 1 ist, die die Menge an übriggebliebenem Leben zu diesem Zeitpunkt darstellt (0 = Lebensleiste ist leer, 1 = Lebensleiste ist gefüllt) |
+| Long | Zeitstempel ([Windows Ticks](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.ticks)) |
```

Comment:
The Microsoft link is available in German, however, these are often machine-translated, so I would not recommend linking to them.

Reply:
I agree. There *might* be a better source for German readers to read about this topic (as in: not the Microsoft documentation), but I can't think of one off the top of my head.